### PR TITLE
feat: session id and device id

### PR DESCRIPTION
**Pull request by merlinpaypal**
## Description

- Add session ID and device ID to Logger and setGlobalAnalytics
- Add ignore script to build steps
- Update error / api tests for better coverage
- Fix some warnings
- Update test coverage
- Make setGlobalAnalytics static
- Use a UUID as the default session ID

## Testing instructions

- Add setGlobalAnalytics to demo
- Run demo
- Check logs to see that device ID and session ID are passed in

### CODE REVIEW

File: library/src/main/java/com/paypal/messages/logger/Logger.kt [hunk 1]
```diff
@@ -54,9 +58,9 @@ class Logger private constructor() {
 			partnerAttributionId = null,
 			// merchantProfileHash will be later defined by pulling from LocalStorage
 			merchantProfileHash = null,
-			deviceId = Settings.Secure.ANDROID_ID,
+			deviceId = if (deviceId == "") Settings.Secure.ANDROID_ID else deviceId,
 			// TODO Determine SessionId for Logger
-			sessionId = "random_session_id",
+			sessionId = if (deviceId == "") "random_session_id" else sessionId,
```
Comment (grablack) [inline library/src/main/java/com/paypal/messages/logger/Logger.kt]: Should this be if ( sessionId == "" ) ? Also should we keep the random_session_id in there?

File: library/src/main/java/com/paypal/messages/logger/Logger.kt [hunk 2]
```diff
@@ -74,13 +78,27 @@ class Logger private constructor() {
 	}
 
 	fun setGlobalAnalytics(
```
Comment (grablack) [inline library/src/main/java/com/paypal/messages/logger/Logger.kt]: Did you decided against adding setGlobalAnalytics to the companion object ?

File: library/src/main/java/com/paypal/messages/logger/Logger.kt [hunk 3]
```diff
@@ -54,9 +58,9 @@ class Logger private constructor() {
 			partnerAttributionId = null,
 			// merchantProfileHash will be later defined by pulling from LocalStorage
 			merchantProfileHash = null,
-			deviceId = Settings.Secure.ANDROID_ID,
+			deviceId = if (deviceId == "") Settings.Secure.ANDROID_ID else deviceId,
```
Comment (grablack) [inline library/src/main/java/com/paypal/messages/logger/Logger.kt]: I believe the setting.secure is used like this `Settings.Secure.getString(context.contentResolver, Settings.Secure.ANDROID_ID)`

I'm seeing android_id returned with just Settings.Secure.ANDROID_ID

File: library/src/main/java/com/paypal/messages/logger/Logger.kt (hunk 1 above)
Comment (merlinpaypal) [reply library/src/main/java/com/paypal/messages/logger/Logger.kt]: Fixed. Replaced with a UUID

File: library/src/main/java/com/paypal/messages/logger/Logger.kt (hunk 2 above)
Comment (merlinpaypal) [reply library/src/main/java/com/paypal/messages/logger/Logger.kt]: Fixed. Created a GlobalAnalytics object that will be used for static purposes

File: library/src/main/java/com/paypal/messages/logger/Logger.kt [hunk 4]
```diff
@@ -95,26 +83,30 @@ class Logger private constructor() {
 		// If we have an active call, we need to pull the current payload and modify it
 
 		// Check for an existing component payload for this component
-		val oldComponent = this.payload?.components?.find { it.instanceId == component.instanceId }
+		this.payload?.run {
+			val oldComponent = components.find { it.instanceId == component.instanceId }
```
Comment (grablack) [inline library/src/main/java/com/paypal/messages/logger/Logger.kt]: We can combine the 2 ifs by just using indexOfFirst

File: library/src/test/java/com/paypal/messages/config/message/PayPalMessageConfigTest.kt [hunk 5]
```diff
@@ -25,31 +26,34 @@ class PayPalMessageConfigTest {
...
 	fun testSetGlobalAnalytics() {
 		val name = "integration_name"
 		val version = "integration_version"
+		val deviceId = "device_id"
+		val sessionId = "session_id"
+
+		assertEquals("", GlobalAnalytics.integrationName)
+		assertEquals("", GlobalAnalytics.integrationVersion)
+		assertEquals("", GlobalAnalytics.deviceId)
+		assertEquals("", GlobalAnalytics.sessionId)
```
Comment (grablack) [inline library/src/test/java/com/paypal/messages/config/message/PayPalMessageConfigTest.kt:39]: These asserts are also in GlobalAnalyticsTest.kt and in the test testSetGlobalAnalyticsWithNoValues. We should be safe keep only 1 copy

**Review by grablack — APPROVED** [review]

**Pull request merged**